bcitflex/course.py

```diff
@@ -1,36 +1,30 @@
 """Course Blueprint"""
 
 from flask import Blueprint, render_template, request
-from sqlalchemy import select
-from werkzeug.datastructures import ImmutableMultiDict
+from sqlalchemy import not_, select
 
 from bcitflex.app_functions import ModelFilter
-from bcitflex.app_functions.course_query import Match
 from bcitflex.db import DBSession
 from bcitflex.model import Course, Meeting, Offering, Subject, Term
+from bcitflex.model.offering import NOT_AVAILABLE
 
 bp = Blueprint("course", __name__)
 
 
-def filters_from_form(form: ImmutableMultiDict) -> ModelFilter:
-    """Return a list of CourseFilters from the given form."""
-    filters = ModelFilter(Course)
-    filters.add_condition("term_id", form.get("term"), Offering)
-    filters.add_condition("subject_id", form.get("subject"))
-    filters.add_condition("campus", form.get("campus"), Offering)
-    filters.add_condition("code", form.get("code"))
-    filters.add_condition("name", form.get("name"), None, Match.PARTIAL)
-    filters.add_condition("is_available", form.get("available") == "True" or None)
-    return filters
-
-
 @bp.route("/")
 @bp.route("/courses", methods=["GET", "POST"])
 def index():
-    courses = DBSession.scalars(select(Course)).all()
+    filters = ModelFilter(Course)
     if request.method == "POST":
```

Comment:
@jonbiemond  could you make it with GET ? There is really no need of POST

Reply:
Hey @MikeVarchola, I made the change. The tests pass but when I run the dev server I can't get the filters to work. Do you know what I might be doing wrong?

Reply:
try to use this `request.args.get` instead of `request.form.get`
also you need to pass filters in queryParams like I did